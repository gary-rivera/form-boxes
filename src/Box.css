.Box {
    margin-top: 10px;
    display: flex;
    flex-direction: column;
    align-items: left;
}

.Box-button {
    max-width: 100px;
}