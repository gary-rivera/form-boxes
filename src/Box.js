import React from 'react';
import "./Box.css";

function Box({ height=50, width=50, backgroundColor='blue', removeBox }) {
  
  return (
    <div className="Box">
      <div style={{ height, width, backgroundColor }}></div>
      <button className="Box-button" onClick={removeBox}>Remove The Box!</button>
    </div>
  )
}

export default Box;