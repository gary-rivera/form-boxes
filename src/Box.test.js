import { fireEvent, render } from '@testing-library/react'
import Box from './Box'


it("renders without crashing", function () {
    render(<Box />)
})

it("matches default snapshot", function () {
    const { container } = render(<Box />)
    expect(container).toMatchSnapshot();
})

it("matches modified box snapshot", function () {
    const { container } = render(
        <Box
            height='25px'
            width='25px'
            backgroundColor= 'red'
        />)
    expect(container).toMatchSnapshot();
})
