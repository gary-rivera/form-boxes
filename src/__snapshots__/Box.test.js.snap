// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches default snapshot 1`] = `
<div>
  <div
    class="Box"
  >
    <div
      style="height: 50px; width: 50px; background-color: blue;"
    />
    <button
      class="Box-button"
    >
      Remove The Box!
    </button>
  </div>
</div>
`;

exports[`matches modified box snapshot 1`] = `
<div>
  <div
    class="Box"
  >
    <div
      style="height: 25px; width: 25px; background-color: red;"
    />
    <button
      class="Box-button"
    >
      Remove The Box!
    </button>
  </div>
</div>
`;
